### PR TITLE
Running Unit test on macOS

### PR DESCRIPTION
**Pull request by prietyc123**
**What type of PR is this?**

king/feature

**What does does this PR do / why we need it**:

Run Unit test on macOs

**Which issue(s) this PR fixes**:

Fixes - Part of #2539 

**How to test changes / Special notes to the reviewer**:

test should run on macOS in travis CI

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -28,6 +28,11 @@ jobs:
         # submit coverage.txt to codecov.io
         - bash <(curl -s https://codecov.io/bash)
 
+    - &mac-test
```
Comment (mik-dass) [inline .travis.yml]: What is this test for?

Comment (prietyc123) [reply .travis.yml]: > What is this test for?

YAML alias for sharing the settings across unit test on macOS

Comment (mik-dass) [reply .travis.yml]: ok

**Comment by mik-dass** [comment]
@prietyc123 Do we need this job https://travis-ci.com/openshift/odo/jobs/281842488? It seems to do nothing. Am I missing something?

**Comment by prietyc123** [comment]
> @prietyc123 Do we need this job https://travis-ci.com/openshift/odo/jobs/281842488? It seems to do nothing. Am I missing something?

Fixing this 

/hold

**Comment by amitkrout** [comment]
```
E0131 05:08:03.195040    5465 watch.go:183] Failed getting details of the changed file /var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/fabric8-analytics-license-analysis372271615/tests216276105/test_read_licenses.py563585660. Ignoring the change

E0131 05:08:03.197046    5465 watch.go:189] Ignoring event for file /var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/fabric8-analytics-license-analysis372271615/tests216276105/test_read_licenses.py563585660 as details about the file couldn't be fetched

--- FAIL: TestWatchAndPush (9.48s)
[...]
```
I can see the same failure locally too. @mik-dass @prietyc123 Can you please have a look

**Comment by mik-dass** [comment]
> I can see the same failure locally too. @mik-dass @prietyc123 Can you please have a look

I don't have a mac to test and debug the failure. On linux it works fine though.

**Comment by prietyc123** [comment]
> I can see the same failure locally too. @mik-dass @prietyc123 Can you please have a look

I am also getting the same error locally. I will debug and get the info

**Comment by amitkrout** [comment]
> > I can see the same failure locally too. @mik-dass @prietyc123 Can you please have a look
> 
> I am also getting the same error locally. I will debug and get the info

@prietyc123 While debugging i tried multiple time with ```make test``` and surprisingly i am not able to hit the issue. However travis consistently hitting the same failure issue. cc_ @mik-dass

**Comment by prietyc123** [comment]
Please share your input on #2551 

/hold

**Comment by amitkrout** [comment]
@prietyc123 We do not need code coverage report here - https://travis-ci.com/openshift/odo/jobs/282233648#L187 as its already been published through https://travis-ci.com/openshift/odo/jobs/282233641#L572. You need to handle this in ```.travis.yaml ```

**Comment by prietyc123** [comment]
Ping @kadel @amitkrout 
PR is ready for review

File: .travis.yml [hunk 2]
```diff
@@ -103,11 +120,4 @@ jobs:
...
     #   script:
-    #     ./scripts/test-install.sh
-    # test installation script on macOS
-    # - state: test
-    #   os: osx
-    #   install:
-    #     - true
-    #   script:
-    #     ./scripts/installer.sh
+    #     ./scripts/test-install.sh
```
Comment (amitkrout) [inline .travis.yml]: @prietyc123 May be you need to remove this commented line.

File: .travis.yml [hunk 3]
```diff
@@ -28,6 +28,23 @@ jobs:
         # submit coverage.txt to codecov.io
         - bash <(curl -s https://codecov.io/bash)
 
+    # YAML alias, for settings shared across the tests on macOS
+    - &osx-test
+      stage: Test
```
Comment (girishramnani) [inline .travis.yml]: is osX tests doing anything different then the linux tests?

Comment (prietyc123) [reply .travis.yml]: > is osX tests doing anything different then the linux tests?

No it is just running the unit test on mac but we don't want https://github.com/openshift/odo/pull/2549/files#diff-354f30a63fb0907d4ad57269548329e3L29 to be done in mac test. This was becoming duplication earlier as mentioned by @amitkrout in comment https://github.com/openshift/odo/pull/2549#issuecomment-580772298 .

Again we want to trigger a job for macOS where unit, integration and e2e test will run. So just created a separate template for osx. Like we are using one for linux.

File: .travis.yml [hunk 4]
```diff
@@ -13,7 +13,7 @@ jobs:
     - &base-test
       stage: test
       go_import_path: github.com/openshift/odo
-      go: "1.11.2"
+      go: "1.13.1"
```
Comment (amitkrout) [inline .travis.yml]: +1 as per https://github.com/openshift/odo/issues/2551#issuecomment-580777174

File: .travis.yml [hunk 5]
```diff
@@ -28,6 +28,23 @@ jobs:
...
+      go_import_path: github.com/openshift/odo
+      go: "1.13.1"
+      install:
+        - make goget-tools
+      script:
+        - export PATH="$PATH:$GOPATH/bin"
+        - make bin
+        - make validate
+        - make test
+      after_success: skip
```
Comment (amitkrout) [inline .travis.yml]: +1, Thanks for following the semantic

**Pull request merged**